File: go_src/6.go
func convert(s string, numRows int) string {
	n := len(s)
	if n < 3 || numRows == 1 {
		return s
	}
	index, direction := 0, 1
	container := make([]string, numRows)
	for i := 0; i < n; i++ {
		container[index] += string(s[i])
		index += direction
		if index == 0 || index == numRows - 1 {
			direction *= -1
		}
	}
	result := ""
	for _, c := range container {
		result += c
	}

	return result
}

File: java_src/6.java
class Solution {
    public String convert(String s, int numRows) {
        if (s == null || s.length() < 2 || numRows == 1) {
            return s;
        }
        StringBuilder[] array = new StringBuilder[numRows];
        for (int i = 0; i < numRows; i++) {
            array[i] = new StringBuilder();
        }
        int index = 0, direction = 1;
        for (char c: s.toCharArray()) {
            array[index].append(c);
            index += direction;
            if (index == 0 || index == numRows - 1) {
                direction *= -1;
            }
        }
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < numRows; i++) {
            result.append(array[i]);
        }
        return result.toString();
    }

    public static void main(String[] args) {
        Solution slo = new Solution();
        System.out.println(slo.convert("ABCDEFG", 3));
    }
}
